### PR TITLE
Adding the support for secret version delayed destroy in the terraform provider

### PR DESCRIPTION
**Pull request by gptSanyam**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
secretmanager: added `version_destroy_ttl` field to `google_secret_manager_secret` resource
```

### CODE REVIEW

**Review by c2thorn — CHANGES_REQUESTED** [review]
Can you add two changelog notes for the two new fields following https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/

Also, there is some trailing whitespace in one of the added YAML lines that our linter is picking up

File: mmv1/third_party/terraform/services/secretmanager/resource_secret_manager_secret_test.go.erb [hunk 1]
```diff
@@ -1104,6 +1156,55 @@ resource "google_secret_manager_secret" "secret-basic" {
...
+      replicas {
+        location = "us-central1"
+      }
+      replicas {
+        location = "us-east1"
+      }
+    }
+  }
+
+  ttl = "86400s"
```
Comment (c2thorn) [inline mmv1/third_party/terraform/services/secretmanager/resource_secret_manager_secret_test.go.erb:1202]: Was this supposed to be `version_destroy_ttl` instead?

File: mmv1/third_party/terraform/services/secretmanager/resource_secret_manager_secret_test.go.erb [hunk 2]
```diff
@@ -379,6 +379,58 @@ func TestAccSecretManagerSecret_ttlUpdate(t *testing.T) {
...
+				Config: testAccSecretManagerSecret_withoutVersionDestroyTtl(context),
+			},
+			{
+				ResourceName:            "google_secret_manager_secret.secret-basic",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"ttl", "labels", "terraform_labels"},
+			},
+			{
+				Config: testAccSecretManagerSecret_basic(context),
```
Comment (c2thorn) [inline mmv1/third_party/terraform/services/secretmanager/resource_secret_manager_secret_test.go.erb:404]: Will this step be needed in between `testAccSecretManagerSecret_withoutVersionDestroyTtl` and `testAccSecretManagerSecret_versionDestroyTtlUpdate` ? That's fine is so, but just checking

**Review by c2thorn — CHANGES_REQUESTED** [review]
Need confirmation on the update test

File: mmv1/third_party/terraform/services/secretmanager/resource_secret_manager_secret_test.go.erb (hunk 1 above)
Comment (gptSanyam) [reply mmv1/third_party/terraform/services/secretmanager/resource_secret_manager_secret_test.go.erb:1202]: Thanks for pointing this, will update this.

File: mmv1/third_party/terraform/services/secretmanager/resource_secret_manager_secret_test.go.erb (hunk 2 above)
Comment (gptSanyam) [reply mmv1/third_party/terraform/services/secretmanager/resource_secret_manager_secret_test.go.erb]: Removing this condition.

**Review by c2thorn — APPROVED** [review]
LGTM, thanks for the changes

**Pull request merged**